src/components/NewDatePicker/CalendarPicker/index.js

```diff
@@ -2,200 +2,103 @@ import _ from 'underscore';
-    }
-
     /**
      * Calls the onSelected function with the selected date.
      * @param {Number} day - The day of the month that was selected.
      */
     onDayPressed(day) {
         this.setState(
-            (prev) => {
-                const momentDate = moment(prev.currentDateView).date(day);
-
-                return {
-                    selectedDay: this.getNumberStringWithLeadingZero(day),
-                    currentDateView: this.clampDate(momentDate.toDate()),
-                };
-            },
-            () => {
-                this.props.onSelected(this.getSelectedDateString());
-            },
+            (prev) => ({
+                currentDateView: moment(prev.currentDateView).set('date', day).toDate(),
+            }),
+            () => this.props.onSelected(moment(this.state.currentDateView).format('YYYY-MM-DD')),
         );
     }
 
-    /**
-     * Gets the date string build from state values of selected year, month and day.
-     * @returns {string} - Date string in the 'YYYY-MM-DD' format.
-     */
-    getSelectedDateString() {
-        // can't use moment.format() method here because it won't allow incorrect dates
-        return `${this.state.selectedYear}-${this.state.selectedMonth}-${this.state.selectedDay}`;
-    }
-
-    /**
-     * Returns the string converted from the given number. If the number is lower than 10,
-     * it will add zero at the beginning of the string.
-     * @param {Number} number - The number to be converted.
-     * @returns {string} - Converted string prefixed by zero if necessary.
-     */
-    getNumberStringWithLeadingZero(number) {
-        return `${number < 10 ? `0${number}` : number}`;
-    }
-
-    /**
-     * Gives the new version of the state object,
-     * changing both selected month and year based on the given moment date.
-     * @param {moment.Moment} momentDate - Moment date object.
-     * @returns {{currentDateView: Date, selectedMonth: string, selectedYear: string}} - The new version of the state.
-     */
-    getMonthState(momentDate) {
-        const clampedDate = this.clampDate(momentDate.toDate());
-        const month = clampedDate.getMonth() + 1;
-
-        return {
-            selectedMonth: this.getNumberStringWithLeadingZero(month),
-            selectedYear: clampedDate.getFullYear().toString(), // year might have changed too
-            currentDateView: clampedDate,
-        };
-    }
-
     /**
      * Handles the user pressing the previous month arrow of the calendar picker.
      */
     moveToPrevMonth() {
-        this.setState(
-            (prev) => {
-                const momentDate = moment(prev.currentDateView).subtract(1, 'M');
-
-                return this.getMonthState(momentDate);
-            },
-            () => {
-                this.props.onSelected(this.getSelectedDateString());
-            },
-        );
+        this.setState((prev) => ({currentDateView: moment(prev.currentDateView).subtract(1, 'M').toDate()}));
```

Comment:
```suggestion
        this.setState((prev) => ({currentDateView: moment(prev.currentDateView).subtract(1, 'months').toDate()}));
```

Maybe `months` to be more concise with what you did on `moment(prev.currentDateView).set('year', year).toDate()` for example?